### PR TITLE
Fix issue #86

### PR DESCRIPTION
**Pull request by xiaomi7732**
Fix for #86. Speed up function to parse json. 85 times faster gain for the get_jsons method by simple benchmark.

Details:
Bench mark:
10000 iterations, 0.350833 / 0.004146 = 84.619633. (~85 times faster).
Details:

  Start of content (New)
  {"time":"2017-08-24T21:00:23.5420000Z","systemId":"e22a0996-e5a7-4952-8e28-4357a6e8f0c5","category":"
  0.010000   0.000000   0.010000 (  0.004146)

  saars@DESKTOP-F8RTJFM:/mnt/d/Ruby/azure-ruby$ ruby getjson.2.rb
  Start of content (Benchmark)
  {"time":"2017-08-24T21:00:23.5420000Z","systemId":"e22a0996-e5a7-4952-8e28-4357a6e8f0c5","category":"
  0.350000   0.000000   0.350000 (  0.350833)

### CODE REVIEW

File: Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb [hunk 1]
```diff
@@ -207,46 +207,46 @@ def process(queue)
...
+      count += 1
+      move_opening = true
+      move_closing = false
+    elsif content[index] == '}'
+      count -= 1
+      move_closing = true
+      move_opening = false
+    end #if
+    index += 1
+    hit += 1 if count == 0
```
Comment (clguimanMSFT) [inline Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: shouldn't there also be a check for malformed json when count < 0 ?

File: Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb [hunk 2]
```diff
@@ -207,46 +207,46 @@ def process(queue)
...
+  return nil if (content.index '{').nil?
+
+  hit = 0
+  count = 0
+  index = 0
+  first = content.index('{')
+  move_opening = true
+  move_closing = true
+  while(hit < batch_size)
+    inIndex = content.index('{', index) if move_opening
```
Comment (clguimanMSFT) [inline Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: don't forget to exclude '{' and '}' inside string values

Comment (xiaomi7732) [reply Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: Good feedback. That scenario is tracked by #85. I prefer to deal with it separately.

**Comment by xiaomi7732** [comment]
@clguimanMSFT,
Thanks for the quick turn-around

**Pull request merged**